src/qtism/runtime/processing/ResponseProcessingEngine.php

```diff
@@ -161,19 +161,26 @@ protected function &getTemplateMapping()
     public function process()
     {
         $rules = $this->getResponseProcessingRules();
+        $exception = null;
 
-        try {
-            foreach ($rules as $rule) {
+        foreach ($rules as $rule) {
+            try {
                 $engine = new RuleEngine($rule, $this->getContext());
                 $engine->process();
                 $this->trace($rule->getQtiClassName() . ' executed');
+            } catch (\Throwable $exception) {
+                if ($exception instanceof RuleProcessingException
```

Comment:
It will ignore other exceptions which occur earlier, right?

Reply:
You're right, only the last exception would be thrown. We'll lose all previous ones.

Reply:
Is this an issue? I mean the point of these exception is to raise the fact there was an issue during this process

Also this is already what is happening now, but instead of throwing only the last one, we throw the first one

Reply:
It would be better to gather all the exceptions into one specific kind of exception, so that qti-sdk is just giving all the information about what is done and what is not, and the calling code to do whatever they want with it (i.e. log the different exceptions then keep the test going or stopping). Most importantly, the calling code must be aware that the computed variables (E.G. SCORE, MAXSCORE, ...) are not relevant and scoring has to be redone afterwards with fixed rules. Having the exact feedback of every wrongly written rule is then the best help we can get from this.